### PR TITLE
Add icons to events

### PR DESCRIPTION
**Pull request by kubaolszewski**

### CODE REVIEW

File: lib/app/custom_widgets/event_text_with_icon.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    final eventType = event?.type ?? AppConstVariables.stringPlaceholder;
+    final eventDetail = event?.detail ?? AppConstVariables.stringPlaceholder;
+    final playerIn = assist ?? AppConstVariables.stringPlaceholder;
+    return Row(
+      mainAxisAlignment: isHomeTeam == true ? MainAxisAlignment.start : MainAxisAlignment.end,
+      children: [
+        if (isHomeTeam == true) ...[
```
Comment (jakgrab) [inline lib/app/custom_widgets/event_text_with_icon.dart]: `isHometeam` is not nullable, change to if (isHomeTeam) ... []

File: lib/app/custom_widgets/event_text_with_icon.dart [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          Text(
+            "$time'",
+            style: const CustomTextStyle(fontSize: 12, color: AppColors.inactiveTextGrey, fontWeight: FontWeight.w700),
+          ),
+          const SizedBox(width: 6),
+          if (eventType == AppConstVariables.goal)
```
Comment (jakgrab) [inline lib/app/custom_widgets/event_text_with_icon.dart]: Maybe you can convert these ifs to switch statement? Looks messy tbh

File: lib/app/custom_widgets/event_text_without_icon.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  final String time;
+  final String player;
+  final bool isHomeTeam;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisAlignment: isHomeTeam == true ? MainAxisAlignment.start : MainAxisAlignment.end,
+      children: [
+        isHomeTeam == true
```
Comment (jakgrab) [inline lib/app/custom_widgets/event_text_without_icon.dart]: Again `isHomeTeam` ? ... : ...

File: lib/app/custom_widgets/events_widgets/card_event_widget.dart [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  final String eventDetail;
+  final String player;
+  final bool isHomeTeam;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        if (isHomeTeam == true) ...[
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/card_event_widget.dart]: Again here

File: lib/app/custom_widgets/events_widgets/card_event_widget.dart [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            children: [
+              if (eventDetail == AppConstVariables.yellowCard) ...[
+                Align(
+                  alignment: Alignment.centerLeft,
+                  child: Text(player,
+                      style: const CustomTextStyle(fontSize: 12, color: Colors.white, fontWeight: FontWeight.w700)),
+                ),
+                const SizedBox(width: 12),
+                const Image(
+                  image: AssetImage('${AppConstVariables.assetsPath}yellow_card.png'),
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/card_event_widget.dart]: Why not store the whole asset path in `AppConstVariables` e.g.  in class `AppConstVariables` add `static const yellowCard = 'assets/yellow_card.png';`

File: lib/app/custom_widgets/events_widgets/card_event_widget.dart [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                ),
+              ] else if (eventDetail == AppConstVariables.redCard) ...[
+                Align(
+                  alignment: Alignment.centerLeft,
+                  child: Text(player,
+                      style: const CustomTextStyle(fontSize: 12, color: Colors.white, fontWeight: FontWeight.w700)),
+                ),
+                const SizedBox(width: 12),
+                const Image(
+                  image: AssetImage('${AppConstVariables.assetsPath}red_card.png'),
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/card_event_widget.dart]: Same here: `static const redCard = 'assets/red_card.png'`.

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+  final String playerOut;
+  final bool isHomeTeam;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        if (isHomeTeam == true) ...[
+          const Image(
+            image: AssetImage(('${AppConstVariables.assetsPath}substitution.png')),
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: Same here, you can create a separate `AssetsPaths` class and store these there.

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Expanded(
+                  child: Align(
+                    alignment: Alignment.centerLeft,
+                    child: Row(
+                      children: [
+                        const Text(
+                          'In: ',
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: Extract this to l10n

File: lib/app/custom_widgets/event_text_with_icon.dart (hunk 1 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/event_text_with_icon.dart]: you're totally right

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+                      ],
+                    ),
+                  ),
+                ),
+                Expanded(
+                  child: Align(
+                    alignment: Alignment.centerLeft,
+                    child: Row(
+                      children: [
+                        const Text('Out: ',
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: Same here, extract this string

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Expanded(
+                  child: Align(
+                    alignment: Alignment.centerLeft,
+                    child: Row(
+                      children: [
+                        const Text(
+                          'In: ',
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: And here

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+                      ],
+                    ),
+                  ),
+                ),
+                Expanded(
+                  child: Align(
+                    alignment: Alignment.centerLeft,
+                    child: Row(
+                      children: [
+                        const Text('Out: ',
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: here also

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+                      ],
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ),
+          const SizedBox(width: 12),
+          const Image(
+            image: AssetImage(('${AppConstVariables.assetsPath}substitution.png')),
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: Again, export the assets and store paths in variables

File: lib/app/custom_widgets/events_widgets/var_event_widget.dart [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+  final String player;
+  final bool isHomeTeam;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        if (isHomeTeam == true) ...[
+          Image(
+            image: const AssetImage(('${AppConstVariables.assetsPath}var.png')),
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/var_event_widget.dart]: Same here

File: lib/app/custom_widgets/events_widgets/var_event_widget.dart [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+          Text(player,
+              style:
+                  const CustomTextStyle(fontSize: 12, color: AppColors.inactiveTextGrey, fontWeight: FontWeight.w700))
+        ] else ...[
+          Text(player, style: const CustomTextStyle(fontSize: 12, color: Colors.white, fontWeight: FontWeight.w700)),
+          const SizedBox(
+            width: 12,
+          ),
+          Image(
+            image: const AssetImage(('${AppConstVariables.assetsPath}var.png')),
```
Comment (jakgrab) [inline lib/app/custom_widgets/events_widgets/var_event_widget.dart]: And here

File: lib/app/custom_widgets/event_text_with_icon.dart (hunk 2 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/event_text_with_icon.dart]: I tried to but I think I would have to either rewrite `eventType` variable in a different way or replace const variables used here with some kind of enum, I will try to work it out now

Comment (jakgrab) [reply lib/app/custom_widgets/event_text_with_icon.dart]: Oh I haven't realised that's not an enum, it should be!

File: lib/app/custom_widgets/event_text_without_icon.dart (hunk 3 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/event_text_without_icon.dart]: done

File: lib/app/custom_widgets/events_widgets/card_event_widget.dart (hunk 4 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/card_event_widget.dart]: done

File: lib/app/custom_widgets/events_widgets/card_event_widget.dart (hunk 5 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/card_event_widget.dart]: done in a way recommended in next review with separate class

File: lib/app/custom_widgets/events_widgets/card_event_widget.dart (hunk 6 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/card_event_widget.dart]: done

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart (hunk 7 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: right, done that

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart (hunk 8 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: done

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart (hunk 9 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: done

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart (hunk 10 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: done

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart (hunk 11 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: done

File: lib/app/custom_widgets/events_widgets/substitution_event_widget.dart (hunk 12 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/substitution_event_widget.dart]: done

File: lib/app/custom_widgets/events_widgets/var_event_widget.dart (hunk 13 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/var_event_widget.dart]: done

File: lib/app/custom_widgets/events_widgets/var_event_widget.dart (hunk 14 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/events_widgets/var_event_widget.dart]: done

**Review by jakgrab — APPROVED** [review]

File: lib/app/custom_widgets/event_text_with_icon.dart (hunk 2 above)
Comment (kubaolszewski) [reply lib/app/custom_widgets/event_text_with_icon.dart]: Came up with something, was quite difficult considering I had to compare string with enum, so as a result I had to convert enums to strings and wrote a bunch of extra code but so far it works as good as before

**Pull request merged**